python/pyspark/mllib/linalg.py

```diff
@@ -208,9 +209,55 @@ def __init__(self, ar):
             ar = ar.astype(np.float64)
         self.array = ar
 
+    def toString(self):
+        """
+        Convert DenseVector to string representation.
+
+        >>> a = DenseVector([0, 1, 2, 3])
+        >>> a.toString()
+        '[0.0,1.0,2.0,3.0]'
+        """
+        return str(self)
+
+    def copy(self):
+        return DenseVector(np.copy(self.array))
+
+    @staticmethod
+    def parse(vectorString):
+        """
+        Parse string representation back into the DenseVector.
+
+        >>> DenseVector.parse('[0.0,1.0,2.0,3.0]')
+        DenseVector([0.0, 1.0, 2.0, 3.0])
+        """
+        vectorString = vectorString[1:-1]
```

Comment:
I think the Scala version is a bit more robust since it uses Java's StringTokenizer for splitting